pkg/scheduler/scheduler_test.go

```diff
@@ -916,6 +900,14 @@ func setupTestScheduler(ctx context.Context, queuedPodStore *clientcache.FIFO, s
 		recorder = &events.FakeRecorder{}
 	}
 
+	var schedulingQueue *queue.PriorityQueue
+	if informerFactory == nil {
+		informerFactory = informers.NewSharedInformerFactory(fake.NewSimpleClientset(), 0)
```

Comment:
I don't think we need to create informerFactory. if it is nil, let the test fail. WDYT?

Reply:
I think we can remove `informerFactory` from the function signature; instead, instantiate an informerFactory unconditionally here, and return it for the caller to consume.

(BTW: and check if we can eliminate `NewTestQueue`)

Reply:
> I don't think we need to create informerFactory here. if it is nil, let the test fail. WDYT?

Aha, made a mistake here, it was not my really intension. Sorry.

Reply:
> I think we can remove `informerFactory` from the function signature; instead, instantiate an informerFactory unconditionally here, and return it for the caller to consume.

 We have some places build informer before: https://github.com/kubernetes/kubernetes/blob/834349ed1979073b4f6ed3e16393d31c19008701/pkg/scheduler/scheduler_test.go#L966-L977
> (BTW: and check if we can eliminate `NewTestQueue`)

`NewTestQueue` used at quite a lot of places and I think it simplified the usage for users to construct a priority queue, so I think we can keep it.